### PR TITLE
fix(agent): improve UUID generation in QA test case prompts

### PR DESCRIPTION
**Pull request by MH4GF**
## Issue

Resolves issue with hardcoded UUID values in generated SQL test cases.

## Why is this change needed?

The QA agent was generating SQL test cases with hardcoded UUID strings like `'11111111-1111-1111-1111-111111111111'` instead of using dynamic UUID generation functions. This caused:

- Static test data that doesn't represent real-world scenarios
- Potential conflicts when running tests multiple times
- Less realistic test coverage

### 1. Prompt Structure Refactoring
- Restructured `SYSTEM_PROMPT_FOR_TESTCASE_GENERATION` into modular components:
  - `ROLE_CONTEXT`: Basic role definition
  - `CRITICAL_INSTRUCTIONS`: Tool usage requirements
  - `TEST_REQUIREMENTS`: Test scenario specifications (with UUID improvement)
  - `EXAMPLES`: Tool structure example
  - `STOP_CONDITIONS`: Completion criteria

### 2. UUID Generation Improvement
- Added instruction: `"Use gen_random_uuid() for UUID columns, not hardcoded strings"`
- Promotes use of PostgreSQL's built-in `gen_random_uuid()` function
- Eliminates hardcoded UUID patterns

### Before (Hardcoded UUIDs):
```sql
INSERT INTO users (id, email) VALUES ('11111111-1111-1111-1111-111111111111','user@example.com');
INSERT INTO tasks (id, user_id, title, status) VALUES ('22222222-2222-2222-2222-222222222222','11111111-1111-1111-1111-111111111111','Sample Task','open');
```

### After (Dynamic UUIDs):
```sql
INSERT INTO users(id, email) VALUES (gen_random_uuid(), 'alice@example.com');

WITH u AS (SELECT id FROM users WHERE email='alice@example.com')
INSERT INTO tasks(id, user_id, title, status) SELECT gen_random_uuid(), id, 'Finish report', 'open' FROM u;
```

## Benefits

- **Dynamic UUID generation** for each test execution
- **More realistic test data** that better represents production scenarios  
- **Improved SQL quality** with better patterns (WITH clauses, subqueries)
- **Maintainable structure** enabling future prompt improvements

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **Refactor**
  * Reworked the test case generation prompt into modular parts for improved reliability and maintainability.
  * Expanded examples and standardized test-case structure (titles, descriptions, operation details) for clearer, more consistent outputs.
  * Added explicit guidance to use gen_random_uuid() for UUID columns, reducing invalid SQL in generated cases.
  * Introduced clearer stop conditions to prevent overly long or off-topic responses.
  * Overall, users should see more accurate, readable, and realistic SQL-focused test cases generated by the assistant.

### CODE REVIEW

File: frontend/internal-packages/agent/src/qa-agent/testcaseGeneration/prompts.ts [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-CRITICAL: You MUST use the saveTestcase tool to save your generated test case. Do not provide test cases as text.
+const CRITICAL_INSTRUCTIONS =
+  'CRITICAL: You MUST use the saveTestcase tool to save your generated test case. Do not provide test cases as text.'
 
+const TEST_REQUIREMENTS = `
 Create both positive and negative test scenarios that include:
 - Valid data insertion tests
 - Constraint violation tests (NOT NULL, FOREIGN KEY, UNIQUE)
 - Edge cases and boundary conditions
+- Use gen_random_uuid() for UUID columns, not hardcoded strings
```
[the remark this answers is not in the document]
Comment (MH4GF) [reply frontend/internal-packages/agent/src/qa-agent/testcaseGeneration/prompts.ts:14]: Due to the Pink Elephant Problem affecting output, it is not included here.

Comment (MH4GF) [inline frontend/internal-packages/agent/src/qa-agent/testcaseGeneration/prompts.ts:14]: This is the one line I added this time.

If the user is using the uuid-ossp extension, it might be better to use `uuid_generate_v4()`, but for now we'll try a small approach. In the future, we might implement an improvement that dynamically changes the prompt based on the loaded extension.

File: frontend/internal-packages/agent/src/qa-agent/testcaseGeneration/prompts.ts [hunk 2]
```diff
@@ -28,12 +30,29 @@ Use the saveTestcase tool with this structure:
...
+${ROLE_CONTEXT}
+
+${CRITICAL_INSTRUCTIONS}
+
+${TEST_REQUIREMENTS}
+
+${EXAMPLES}
+
+${STOP_CONDITIONS}
+`
```
Comment (MH4GF) [inline frontend/internal-packages/agent/src/qa-agent/testcaseGeneration/prompts.ts:54]: To address the issue where LLM prompts often alter the majority of the prompt during prompt engineering, we split instructions into sections for localization. This also makes the prompt structure easier for humans to understand.
This approach has been influenced by the following urls.


- https://github.com/langchain-ai/open-swe/blob/53e5dd954db2c663ff5b4cb3e4ae6c13538456f3/apps/open-swe/src/graphs/manager/nodes/classify-message/prompts.ts
- https://x.com/mattpocockuk/status/1958179930262356032

**Review by hoshinotsuyoshi — APPROVED** [review]
LGTM!
